### PR TITLE
doc: clarify that user must be site admin to generate site credentials

### LINKED ISSUE #606: Add Swagger documentation to `POST` `/sites/{site_id}:generateCredentials` API endpoint
#### Background

According to https://github.com/microbiomedata/nmdc-runtime/issues/605, a team member did not know that the `POST` `/sites/{site_id}:generateCredentials` API endpoint could only be used to generate credentials for a site **the current user is a "site admin" of.**

#### Task

The task here is to add some documentation to that API endpoint so that Swagger UI shows that limitation. The documentation could be an annotation on the `site_id` URL parameter or on the endpoint as a whole.

#### Follow-on task

After adding the aforementioned documentation, add documentation to other API endpoints also. I think this will result in teammates being able to use the API more easily.

### PR DESCRIPTION
**Pull request by dwinston**
Fixes #606 

## Type of change

Please delete options that are not relevant.

- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] visual inspection of generated docs UI

image

# Definition of Done (DoD) Checklist:

- [x] My code follows the style guidelines of this project (have you run `black nmdc_runtime/`?)
- [x] I have performed a self-review of my code

### CODE REVIEW

File: nmdc_runtime/api/endpoints/sites.py [hunk 1]
```diff
@@ -193,10 +193,18 @@ def get_site_object_link(
 
 @router.post("/sites/{site_id}:generateCredentials", response_model=ClientCredentials)
 def generate_credentials_for_site_client(
-    site_id: str,
+    site_id: str = Path(
+        ...,
+        description="The ID of a site for which you, the authenticated user, are an admin.",
```
Comment (eecavanna) [inline nmdc_runtime/api/endpoints/sites.py:198]: Given the new endpoint description, I recommend saying "_the_ site", in reference to the _same site_ as the one you want to generate credentials for. I think the endpoint description sets the stage sufficiently for calling it "the site."

```suggestion
        description="The ID of the site.",
```

File: nmdc_runtime/api/endpoints/sites.py [hunk 2]
```diff
@@ -193,10 +193,18 @@ def get_site_object_link(
...
+    ),
     mdb: pymongo.database.Database = Depends(get_mongo_db),
     user: User = Depends(get_current_active_user),
 ):
+    """
+    Generate a client_id and client_secret for the given site.
+
+    You must be authenticated as a user who is registered as an admin for the given site.
```
Comment (eecavanna) [inline nmdc_runtime/api/endpoints/sites.py:206]:
```suggestion
    You must be authenticated as a user who is registered as an admin of the given site.
```

**Review by eecavanna — APPROVED** [review]
I made a couple suggestions. None of the are deal breakers, from my perspective. Thanks for adding this documentation! Just now catching up with my emails from the day.

**Review by eecavanna — APPROVED** [review]

**Review by aclum — APPROVED** [review]

**Comment by eecavanna** [comment]
I'm comfortable with this being merged into `main`.

**Pull request merged**